### PR TITLE
Cleanup + Integration with HW-accelerated NN

### PR DESCRIPTION
**Pull request by andreamah**
# :ship: Pull Request

## :question: Purpose

The following changes are made:
- interfacing the C hardware neural-network using the `.so` file
- Adding the scripts used the train network and generate `.bin` form of neural network
- Additional training to neural network for better accuracy (will be improved upon in following work most likely)
- Better organization between Tensorflow-based and custom neural network running options
- Consistent running camera in 480p to avoid deviation
- Removed edge detection from default settings for performance reasons
- Misc. cleanup

I am hoping to merge this in before I make more changes to the neural network by potentially basing it less on handwritten letters and more on custom-cropped letters from online photos (as advised in milestone meeting).

## :hammer: Validation Strategy
Custom set of pre-taken photos and manual testing with printed out plates.

**Describe how you verified your changes if unit tests are not sufficient:**

## :tickets: Ticket(s)

Affects #

(Create a copy of that line for each Github Issue affected,
and replace "Affects" with "Closes" if merging this will close the relevant ticket.)

### CODE REVIEW

**Review by steveny9911 — APPROVED** [review]

**Review by Zhang-Kevin16 — APPROVED** [review]
LGTM
